### DISCUSSION TITLE
set the date of death for people

### DISCUSSION THREAD

**Discussion by MotorSportF1** [immich-app/immich#9020] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hello,

Is it possible to include in the calculation the date of death when the person has died?
Because it distorts the calculation of age.

Thanks for all

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by DeltaTango69** [comment]
Are you make photos from ghosts? 😉

**Reply by MotorSportF1** [reply]
Hey, Not but I have already photos from my Grand Father but he is dead since one year.
And the calcul is missing because says 102 years but is dead at 101 years ;)

**Comment by dwantstocode** [comment]
I am also hoping this feature is included/considered in future releases. This is an emotional request.

**Comment by propman07** [comment]
Recommend to add the ability to enter in the date that a person passed away. As a suggestion, when you look at a photo where Immich has identified this person, when it displays their age, Immich would check to see if the date of the image is after the date that the person passed away. If it is, Immich would display the age in parenthesis indicating the age of the person if they were still alive. If the date is before the date that the person passed away, Immich would calculate their age as usual (cool feature, by the way) Ref #18280 as a similar but not identical request / functionality.

**Comment by pducharme** [comment]
+1 on this request.  I have lost someone and would like to have his Date of Death added.

**Comment by waclaw66** [comment]
I don't understand this requirement. Isn't it easier to set the date the photo was taken instead of dealing with the date of death?

**Comment by ShakataGaNai** [comment]
I would like this for pets. Mostly because the models aren't good enough at discerning between pets like they are humans. So its common for similar animals to be tagged the same. It's really sad when a photo pops up and says its your beloved childhood pet, who's been gone for years.

**Comment by ghostc0des** [comment]
I think the request is that if we upload a picture of our great grand parents who have died a long time ago, and we have placed their actual date of birth in immich to something around the year 1820, it will show that the age of the person as approx 200 years old. 

It's not related to the metadata of the image when it was scanned or captured.

<img width="855" height="1024" alt="Screenshot_20260507-100512" src="https://github.com/user-attachments/assets/fe5a3f5f-5bb7-49ea-b075-2be69668675c" />

**Comment by opbillbobaggins** [comment]
I also would like the ability to add date of a person's passing. It could just be a data point viewable when selecting the person. Currently when you select a person it has the ability to add date of birth or it shows date of birth with age but no ability to add their date of passing.

**Comment by Numidiaware** [comment]
Add new fields to person Table:
-deathDate
-marriageDate
-Married since
-age: now-birthDate if deathDate is null else deathDate -  birthDate

-update People card and add a new menus:
-Date of Marriage
-Date of Death
-Details (Includ all person infos )

-add a new formes:
-Date of Marriage
-Date of Death
-Details

**Comment by curljlg** [comment]
Please consider adjusting the birthday feature to account for deceased people. It’s painful to look at a loved one who passed many years ago and see what age they would be if living

**Reply by tim-vk** [reply]
I think the main issue is that dead people are recognized even after their day of death (because of pictures on the background or wrong metadata).

If that would be removed i think you can fix a bunch of the issues people have.
